testing/cython/create_modules/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+#=============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+# TODO: For some reason I have to specify the minimum required cmake version
+# before including init.cmake below even though none of them execute code.
```

Comment:
Ahh yes.

Very crudely, CMake at parse time parses the entire macro and stores a representation of it instead of executing it. That state of all CMake policies are captured at that initial parse and used when the macro is called ( crazy right ). So that is why you need to specify a minimum required version before the macro is parsed